engine/runtime-compiler/src/main/scala/org/enso/compiler/pass/IRPass.scala

```diff
@@ -146,7 +145,7 @@ object IRPass {
       */
     @throws[CompilerError]
     def unsafeAs[T <: Metadata: ClassTag](implicit
-      @unused ev: T =:!= Metadata
+      @unused ev: T =!= Metadata
```

Comment:
Is this a Scala way of doing [Kotlin's reified classes](https://github.com/dukescript/kt-mvvm/blob/master/src/main/java/com/dukescript/api/kt/Models.kt#L17)?

Wouldn't it be simpler to just delete the method implicit magic altogether and just pass in the argument explicitly?

Btw. Why is the method called `unsafeAs` then? Reified classes can be checked during runtime in Kotlin and I see some runtime check here as well.

Reply:
The sole purpose of this logic is to ensure at _compile time_ that neither type inferencer nor explicit type argument result in T being `Metadata` rather than its subclass. 

I don't think providing explicit argument to every usage site is better, plus it doesn't prevent incorrect usage this implicit prevents from.